v4.0.x compile failure under pre C99 standard

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)
v4.0.x branch

## Details of the problem
`mpi.h` contains C99/C++ style comments, `//`, when MPI is compiled with `--enable-mpi1-compatibility`. This breaks compilation under pre C99 standards.

Offending section of `mpi.h`
```
// NOTE: We intentionally do *not* mark the following as
// deprecated/removed because they are used below in function
// prototypes (and would therefore emit warnings, just for #including
// <mpi.h>).
```
Example failure
```
$ cat hello.c
#include "mpi.h"

int main(int argc, char** argv) {
    return 0;
}

$ mpicc -ansi hello.c
/usr/local/openmpi/include/mpi.h:420:1: error: C++ style comments are not allowed in ISO C90
 // NOTE: We intentionally do *not* mark the following as
 ^
/usr/local/openmpi/include/mpi.h:420:1: error: (this will be reported only once per input file)
```

--- a/ompi/include/mpi.h.in
+++ b/ompi/include/mpi.h.in
@@ -417,10 +417,12 @@ typedef ompi_file_errhandler_fn MPI_File_errhandler_fn
 typedef MPI_Win_errhandler_function MPI_Win_errhandler_fn
         __mpi_interface_removed__("MPI_Win_errhandler_fn was removed in MPI-3.0; use MPI_Win_errhandler_function instead");
 
-// NOTE: We intentionally do *not* mark the following as
-// deprecated/removed because they are used below in function
-// prototypes (and would therefore emit warnings, just for #including
-// <mpi.h>).
+/*
+ * NOTE: We intentionally do *not* mark the following as
+ * deprecated/removed because they are used below in function
+ * prototypes (and would therefore emit warnings, just for #including
+ * <mpi.h>).
+ */
 typedef void (MPI_Handler_function)(MPI_Comm *, int *, ...);
 typedef int (MPI_Copy_function)(MPI_Comm, int, void *,
                                 void *, void *, int *);

